core/src/main/scala/org/apache/spark/sql/rapids/tool/AppBase.scala

```diff
@@ -241,11 +241,21 @@ abstract class AppBase(
   protected def checkMetadataForReadSchema(sqlID: Long, planInfo: SparkPlanInfo): Unit = {
     // check if planInfo has ReadSchema
     val allMetaWithSchema = getPlanMetaWithSchema(planInfo)
-    allMetaWithSchema.foreach { node =>
-      val meta = node.metadata
+    allMetaWithSchema.foreach { plan =>
+      val meta = plan.metadata
       val readSchema = ReadParser.formatSchemaStr(meta.getOrElse("ReadSchema", ""))
 
+      val planGraph = SparkPlanGraph(plan)
+      val allNodes = planGraph.allNodes
+      val finalRes = allNodes.filter(node => {
+        // Get ReadSchema of each Node and sanitize it for comparison
+        val trimmedNode = ReadParser.parseReadNode(node).schema.replace("...", "")
+        readSchema.contains(trimmedNode)
+      })
+      val nodeId = if (finalRes.nonEmpty) finalRes.head.id else -1
```

Comment:
have we seen this not match in testing?  Wondering if we really need to parse it twice, vs just using the parseReadNode output

Reply:
This is not relevant as node id is retrieved by different method now.